Realm/Realm.SourceGenerator/Analytics.cs

```diff
@@ -0,0 +1,524 @@
+            foreach (var diagnostic in Diagnostics)
+            {
+                // TODO andrea: this does not up show anywhere. Investigate why.
+                context.ReportDiagnostic(diagnostic);
+            }
+#endif
+        }
+
+        public void AnalyzeRealmClass(ClassInfo classInfo)
+        {
+            if (classInfo.ObjectType == ObjectType.EmbeddedObject)
+            {
+                _realmFeaturesToAnalyse[SdkFeature.IEmbeddedOjbect] = 1;
+            }
+            else if (classInfo.ObjectType == ObjectType.AsymmetricObject)
+            {
+                _realmFeaturesToAnalyse[SdkFeature.IAsymmetricObject] = 1;
+            }
+
+            foreach (var property in classInfo.Properties)
+            {
+                if (property.TypeInfo.IsList)
+                {
+                    if (property.TypeInfo.InternalType.ScalarType == ScalarType.Object)
+                    {
+                        _realmFeaturesToAnalyse[SdkFeature.ReferenceList] = 1;
+                    }
+                    else
+                    {
+                        _realmFeaturesToAnalyse[SdkFeature.PrimitiveList] = 1;
+                    }
+                }
+                else if (property.TypeInfo.IsDictionary)
+                {
+                    if (property.TypeInfo.InternalType.ScalarType == ScalarType.Object)
+                    {
+                        _realmFeaturesToAnalyse[SdkFeature.ReferenceDictionary] = 1;
+                    }
+                    else
+                    {
+                        _realmFeaturesToAnalyse[SdkFeature.PrimitiveDictionary] = 1;
+                    }
+                }
+                else if (property.TypeInfo.IsSet)
+                {
+                    if (property.TypeInfo.InternalType.ScalarType == ScalarType.Object)
+                    {
+                        _realmFeaturesToAnalyse[SdkFeature.ReferenceSet] = 1;
+                    }
+                    else
+                    {
+                        _realmFeaturesToAnalyse[SdkFeature.PrimitiveSet] = 1;
+                    }
+                }
+                else if (property.TypeInfo.IsRealmInteger)
+                {
+                    _realmFeaturesToAnalyse[SdkFeature.RealmInteger] = 1;
+                }
+                else if (property.TypeInfo.TypeSymbol.IsAnyRealmObjectType())
+                {
+                    _realmFeaturesToAnalyse[SdkFeature.RealmObjectReference] = 1;
+                }
+                else if (property.TypeInfo.TypeString == "RealmValue")
+                {
+                    _realmFeaturesToAnalyse[SdkFeature.RealmValue] = 1;
+                }
+            }
+        }
+
+        public void AnalyzeSyntaxNodeForApiUsage(GeneratorSyntaxContext context)
+        {
+            if (context.Node is not IdentifierNameSyntax identifierNameSyntax)
+            {
+                return;
+            }
+
+            switch (identifierNameSyntax.ToString())
```

Comment:
I discussed this already with Andrea, but maybe we can find a solution in which if we already know that a certain feature has already been found we don't need to do a comparison with that string value anymore. 

(We definitely need to have those benchmark running to evaluate the impact of that though)

Reply:
Yes yes, it's a good point. Performance testing needs to be done first and foremost. But improving on this isn't gonna be hard, so I'll likely do it regardless of the performance result.